api/maven-api-core/src/main/java/org/apache/maven/api/services/Source.java

```diff
@@ -20,17 +20,35 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.file.Path;
 
 import org.apache.maven.api.annotations.Experimental;
+import org.apache.maven.api.annotations.Nonnull;
+import org.apache.maven.api.annotations.Nullable;
 
 /**
- * The source for a project's XML model.
+ * The source for parsing an object.
```

Comment:
Is this a Java object being parsed somehow? If not, some synonym would be better

Reply:
Yeah, the wording is not really good.  The idea is that the _thing_ pointed to by the `Source` will be be parsed into.  
I'll fix them to be closer to those from [`org.apache.maven.building.Source`](https://github.com/apache/maven/blob/de19cfcd2bc8e774818d87472e8e64dc37c0b93d/maven-builder-support/src/main/java/org/apache/maven/building/Source.java)